src/ts/ConvivaAnalytics.ts

```diff
@@ -174,35 +185,42 @@ export class ConvivaAnalytics {
     this.debugLog('startsession', this.sessionKey);
   }
 
-  private updateSession = () => {
-    if (!this.sessionDataPopulated) {
-      this.sessionDataPopulated = true;
-      this.updateContentMetadata();
-      this.playerStateManager.updateContentMetadata(this.contentMetadata);
-    }
-  };
-
-  private updateContentMetadata() {
+  /**
+   * Update contentMetadata which must be present before first video frame
+   */
+  private buildContentMetadata() {
     let source = this.player.getConfig().source;
 
-    if (this.contentMetadata.assetName !== this.getAssetName(source)) {
-      this.contentMetadata.assetName = this.getAssetName(source);
-    }
-
+    this.contentMetadata.applicationName = this.config.applicationName || 'Unknown (no config.applicationName set)';
```

Comment:
Should we make `config.applicationName` mandatory? Or even not available outside because this will result in the `Player Name` field on pulse.conviva

Reply:
> Or even not available outside because this will result in the `Player Name` field on pulse.conviva

What do you mean by that? Removing the "unknown" fallback string and only setting it if available?

Reply:
I mean that we could consider making the `applicationName` mandatory in the config. So that we can remove the fallback string. Since `applicationName` will result in the `Player Name` field on pulse i think this should be set always to see were the session is coming from. But it's just my opinion and i'm not 100% sure about that.

Reply:
The fallback is only there to make it obvious that the `applicationName` has not been set (and maybe because it was mandatory in the past) - so this makes it kind of mandatory: if you don't want the unknown warning, set the app name.